assets/js/components/Dashboard/PermanentConfiguration/Workflows/GlEink/StationSelectPage.tsx

```diff
@@ -1,21 +1,30 @@
 import React, { ComponentType, useState } from "react";
-import { WorkflowProps } from "../ConfigureScreensPage";
 import { Container } from "react-bootstrap";
-import PlacesSearchBar, { SearchItem } from "../PlacesSearchBar";
-import { DirectionID } from "../../../../models/direction_id";
-import BottomActionBar from "../BottomActionBar";
+import PlacesSearchBar, { SearchItem } from "../../PlacesSearchBar";
+import WorkflowPlacesList from "../../WorkflowPlacesList";
+import { DirectionID } from "../../../../../models/direction_id";
+import { Place } from "../../../../../models/place";
+import BottomActionBar from "../../BottomActionBar";
 import { useNavigate } from "react-router-dom";
-import WorkflowPlacesList from "../WorkflowPlacesList";
-import { Place } from "../../../../models/place";
 
-const GlEinkWorkflow: ComponentType<WorkflowProps> = ({
+interface StationSelectPageProps {
+  places: Place[];
+  selectedPlaces: Set<string>;
+  setSelectedPlaces: React.Dispatch<React.SetStateAction<Set<string>>>;
+}
+
+const StationSelectPage: ComponentType<StationSelectPageProps> = ({
   places,
-}: WorkflowProps) => {
-  const [selectedPlaces, setSelectedPlaces] = useState<Set<string>>(new Set());
+  selectedPlaces,
+  setSelectedPlaces,
+}: StationSelectPageProps) => {
   const [sortDirection, setSortDirection] = useState<DirectionID>(0);
-  const handleSearchResultClick = (place: SearchItem) => {
+  const handleSearchResultClick = (item: SearchItem) => {
     const existingSelectedPlaces = new Set(selectedPlaces);
-    setSelectedPlaces(existingSelectedPlaces.add(place.id));
+    const placeToAdd = places.find((place) => place.id === item.id);
+    if (placeToAdd) {
+      setSelectedPlaces(existingSelectedPlaces.add(placeToAdd.id));
```

Comment:
Similar issue here, the new state value is "equal" to the previous one since both are references to the same `Set` object, which you mutate to add the new ID.

To make sure react sees that there was a change, you can make a copy for the new state value like so:
```suggestion
      setSelectedPlaces((prev) => new Set([placeToAdd.id, ...prev]));
```

Reply:
I actually did handle that issue with the variable `existingSelectedPlaces` on the line above. That said, I think I do like the extra simplicity of making a new `Set` in the state setter.